### PR TITLE
feat: add millet lsp

### PR DESCRIPTION
**Pull request by igorlfs**
Closes williamboman/mason.nvim#1097.

### CODE REVIEW

File: packages/millet/package.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+name: millet
+description: A language server for Standard ML.
+homepage: https://github.com/azdavis/millet
+licenses:
+  - Apache-2.0
+  - MIT
+languages:
+  - SML
```
Comment (williamboman) [inline packages/millet/package.yaml:9]:
```suggestion
  - Standard ML
```

File: schemas/enums/language.json [hunk 2]
```diff
@@ -126,6 +126,7 @@
         "Shell",
         "Slint",
         "Smithy",
+        "SML",
```
Comment (williamboman) [inline schemas/enums/language.json:129]:
```suggestion
        "Standard ML",
```

**Review by williamboman — COMMENTED** [review]
I think this abbreviation could be expanded to aid discovery, wdyt?

**Comment by igorlfs** [comment]
> I think this abbreviation could be expanded to aid discovery, wdyt?

Sure!

File: schemas/enums/language.json (hunk 2 above)
Comment (williamboman) [inline schemas/enums/language.json:129]:
```suggestion
        "Standard ML",
```

**Review by williamboman — APPROVED** [review]
Great!

**Pull request merged**